refund/refund.js

```diff
@@ -0,0 +1,249 @@
+    const privateKey = bnbCrypto.getPrivateKeyFromMnemonic(mnemonic);
+    this.bnbClient.setPrivateKey(privateKey);
+    try {
+        await this.bnbClient.initChain();
+    } catch (e) {
+        console.log("Error: cannot connect to Binance Chain's lcd server")
+        return
+    }
+
+    // Load our Binance Chain address (required for refunds)
+    const bnbAddrPrefix = network == "mainnet" ? "bnb" : "tbnb"
+    this.bnbChainAddress = bnbCrypto.getAddressFromPrivateKey(privateKey, bnbAddrPrefix);
+
+    return this;
+}
+
+    /**
+     * Manages swap refunds
+     */
+    async refundSwaps() {
+        await this.refundKavaSwaps()
+        await this.refundBinanceChainSwaps()        
+    }
+
+     /**
+     * Refund any expired swaps on Kava
+     */
+    async refundKavaSwaps() {
+        const swapIDs = await this.getRefundableKavaSwaps();
+        console.log(`Kava refundable swap count: ${swapIDs.length}`)
+
+        // Fetch account data so we can manually manage sequence when posting
+        let accountData
+        try {
+            accountData = await Kava.tx.loadMetaData(this.kavaClient.wallet.address, this.kavaClient.baseURI)
+        } catch(e) {
+            console.log(e)
+        }
+
+        // Refund each swap
+        for(let i = 0; i < swapIDs.length; i++) {
+            const sequence = String(Number(accountData.sequence) + i)
+            try {
+                console.log(`\tRefunding swap ${swapIDs[i]}`)
+                const txHash = await this.kavaClient.refundSwap(swapIDs[i], sequence)
+                console.log("\tTx hash:", txHash)
+            } catch (e) {
+                console.log(`\tCould not refund swap ${swapIDs[i]}`)
+                console.log(e)
+            }
+        }
+    }
+
+    /**
+     * Gets the swap IDs of all incoming and outgoing expired swaps on Kava
+     */
+    async getRefundableKavaSwaps() {
+        let expiredSwaps = [];
+        let checkNextBatch = true;
+        let page = 1; // After refunding swaps paginated query results will always start from page 1
+
+        while(checkNextBatch) {
+            let swapBatch;
+            const args = {status: 'Expired', page: page, limit: this.limit};
+            try {
+                swapBatch = await this.kavaClient.getSwaps(5000, args);
+            } catch (e) {
+                console.log(`couldn't query swaps on Kava...`);
+                return
+            }
+            // If swaps in batch, save them and increment page count
+            if(swapBatch.length > 0) {
+                expiredSwaps = expiredSwaps.concat(swapBatch);
+                page++;
+            // If no swaps in batch, don't check the next batch
+            } else {  
+                checkNextBatch = false
+            }
+        }
+    
+        // Calculate each swap's ID as it's not stored in the struct (it's on the interface)
+        let swapIDs = []
+        for(let i = 0; i < expiredSwaps.length; i++) {
```

Comment:
ditto of on the `for...of`

Reply:
https://github.com/Kava-Labs/kava-tools/pull/15/commits/2c9734ae69925fad2bbc85f791a46506316abb9f